File: typescript/src/interfaces/simpleExample.ts
import { PlayerInfo } from "../classes/player";
import Velocity from "../classes/velocity";
import IMove, { MoveResult } from "../iMove";
import { sample } from "lodash";

export default class SimpleMoveExample extends IMove {
    performMove(info: PlayerInfo): MoveResult {
        let velocities = [
            new Velocity(-64, 0), // left
            new Velocity(64, 0),  // right
            new Velocity(0, 64),  // up
            new Velocity(0, -64), // down
        ];
        return { sendMove: true, updateVelocity: sample(velocities) ?? Velocity.Zero };
    }
}

File: dart/lib/interfaces/simple_example.dart
import 'package:lean_dart_client/classes/player.dart';
import 'package:lean_dart_client/classes/velocity.dart';
import 'package:lean_dart_client/move_interface.dart';
import 'package:tuple/tuple.dart';
import 'dart:math';

class SimpleExample extends IMove {
  @override
  Tuple2<bool, Velocity> performMove(PlayerInfo info) {
    var velocities = <Velocity>[
      Velocity.fromXY(-64, 0), // left
      Velocity.fromXY(64, 0), // right
      Velocity.fromXY(0, 64), // up
      Velocity.fromXY(0, -64) // down
    ];
    return Tuple2<bool, Velocity>(
        true, velocities[Random().nextInt(velocities.length - 1)]);
  }
}
